hs/src/Reach/AST/DLBase.hs

```diff
@@ -55,15 +55,19 @@ data DLType
   | T_Object (M.Map SLVar DLType)
   | T_Data (M.Map SLVar DLType)
   | T_Struct [(SLVar, DLType)]
-  | T_TokenBalances Integer
+  | T_Balances
```

Comment:
Because of the way that `fluidVarType` is used, I don't think you actually need to define this, provided you special case it in the `While` part of `Linearize` with the actual type. This will simplify things greatly.